WordPress/Classes/Services/ReaderTopicService.m

```diff
@@ -964,14 +964,19 @@ - (void)mergeMenuTopics:(NSArray *)topics withSuccess:(void (^)(void))success
                         // If the user adds a locally saved tag/interest prevent it from being deleted
                         // while the user is logged out.
                         if ([Feature enabled:FeatureFlagReaderImprovementsPhase2]) {
-                            if (!ReaderHelpers.isLoggedIn && [topic isKindOfClass:ReaderTagTopic.class]) {
-                                ReaderTagTopic *tagTopic = (ReaderTagTopic *)topic;
+                            ReaderTagTopic *tagTopic = (ReaderTagTopic *)topic;
 
-                                if (tagTopic.wasAddedWhileLoggedOut) {
+                            if (!ReaderHelpers.isLoggedIn && [topic isKindOfClass:ReaderTagTopic.class]) {
```

Comment:
Can we not just use tagTopic under this if (tagTopic.title...) here and avoid the kind of class check

Reply:
It looks like that I've added this line but it was already here.

The main issue is that we have different topics: `ReaderTagTopic`, `ReaderListTopic`, `ReaderDefaultTopic`, `ReaderTeamTopic`....

I guess this piece of logic is exclusive to `ReaderTagTopic`.

Reply:
👍